lib/utils.pm

```diff
@@ -534,9 +534,10 @@ Example:
 sub assert_screen_with_soft_timeout {
     my ($mustmatch, %args) = @_;
     # as in assert_screen
-    $args{timeout}             //= 30;
-    $args{soft_timeout}        //= 0;
-    $args{soft_failure_reason} //= $args{bugref} . ': needle(s) not found within ' . $args{soft_timeout};
+    $args{timeout}      //= 30;
+    $args{soft_timeout} //= 0;
+    my $needle_info = ref($mustmatch) eq "ARRAY" ? join(',', @$mustmatch) : $mustmatch;
+    $args{soft_failure_reason} //= $args{bugref} . ': needle(s) ' . $needle_info . ' not found within ' . $args{soft_timeout};
```

Comment:
I think here you can use " for readability

```suggestion
    $args{soft_failure_reason} //= $args{bugref} . ": needle(s) $needle_info not found within " . $args{soft_timeout};
```

Reply:
I don't know if it is just my atom editor that `bugref` is not highlighted in that way in `$args{bugref}` but you're right is more compact...